src/code-generation/CodeGeneratorBase.ts

```diff
@@ -98,14 +98,27 @@ abstract class CodeGeneratorBase<TSettings extends CodeGeneratorBaseSettings> im
     --this._branches.length
     return this
   }
+  wackIndent(oneLine: boolean): this {
+    const brace = this.getOpeningBracket()
+    if (brace) {
+      if (!oneLine || !this.getClosingBracket()) {
+        this.writeLineEnding()
+      }
+    }
 
+    this._branches.push({
+      one_line: oneLine
+    })
+
+    return this
+  }
   protected pushBranch(oneLine: boolean): this {
     const brace = this.getOpeningBracket()
     if (brace) {
       if (!oneLine || !this.getClosingBracket()) {
         this.writeLineEnding()
       }
-      this._result += `${oneLine ? ' ' : this.getIndentation()}${brace}`
+      this._result += `${brace}`
```

Comment:
If Lua forbids there being a space, either add a parameter that indicates a space should not be added by the code generator or override this method in your class.